internal/alb/lb/loadbalancer.go

```diff
@@ -572,6 +441,13 @@ func (l *LoadBalancer) modify(rOpts *ReconcileOptions) error {
 
 // delete Deletes the load balancer from AWS.
 func (l *LoadBalancer) delete(rOpts *ReconcileOptions) error {
+	l.deleted = true
```

Comment:
I think we need to move this to the end, the deleted flag should only be set once the deletion is confirmed, was there a particular reason for moving it up here?

Reply:
i search the code, it's not used in other places(only by my code).

in the end of reconcile part, my code will check l.deleted to choose whether to do "reconcile on association" or not.
If i don't mark it as deleted(maybe shouldBeDeleted) from the beginning, and the delete failed due to low-level errors, my code will tried to reconcile deleted lb(e.g. attach sgs)